packages/flutter/test/widget/navigator_test.dart

```diff
@@ -73,4 +95,18 @@ void main() {
       expect(tester.findText('Y'), isNull);
     });
   });
+
+  test('Navigator.openTransaction fails gracefully when not found in context', () {
+    testWidgets((WidgetTester tester) {
+      StateError exception;
+      Widget widget = new ThirdComponent(onException: (StateError e) {
+        exception = e;
```

Comment:
There's a subtle type error here. You catch all exceptions, but declare that you only accept a StateError. For strict correctness you should probably say the argument is of type `dynamic`.